rest/handler.go

```diff
@@ -258,7 +258,7 @@ func (h *handler) logRequestBody() {
 
 }
 
-func (h *handler) logDuration(realTime bool) {
+func (h *handler) logDuration(realTime bool, err error) {
```

Comment:
I don't like overloading logDuration with conditional error logging - it feels unrelated to the 'logDuration' functionality.  If anywhere it seems like this should be part of h.writeError (which could also reference h.serialNumber if desired).